app/src/main/java/net/demilich/metastone/gui/IconFactory.java

```diff
@@ -48,80 +48,101 @@ public static Image getDialogIcon(DialogType dialogType) {
 
 	public static String getHeroIconUrl(HeroClass heroClass) {
 		String iconPath = RESOURCE_PATH + "/img/heroes/";
-		switch (heroClass) {
-		case DRUID:
-			iconPath += "malfurion";
-			break;
-		case HUNTER:
-			iconPath += "rexxar";
-			break;
-		case MAGE:
-			iconPath += "jaina";
-			break;
-		case PALADIN:
-			iconPath += "uther";
-			break;
-		case PRIEST:
-			iconPath += "anduin";
-			break;
-		case ROGUE:
-			iconPath += "valeera";
-			break;
-		case SHAMAN:
-			iconPath += "thrall";
-			break;
-		case WARLOCK:
-			iconPath += "guldan";
-			break;
-		case WARRIOR:
-			iconPath += "garrosh";
-			break;
-		default:
-		case ANY:
-			iconPath += "unknown";
-			break;
-
+		if (((HeroCard)hero.getSourceCard()).getAsset().equals("<Def>")){
+			switch (hero.getHeroClass()) {
+			case DRUID:
+				iconPath += "malfurion";
+				break;
+			case HUNTER:
+				iconPath += "rexxar";
+				break;
+			case MAGE:
+				iconPath += "jaina";
+				break;
+			case PALADIN:
+				iconPath += "uther";
+				break;
+			case PRIEST:
+				iconPath += "anduin";
+				break;
+			case ROGUE:
+				iconPath += "valeera";
+				break;
+			case SHAMAN:
+				iconPath += "thrall";
+				break;
+			case WARLOCK:
+				iconPath += "guldan";
+				break;
+			case WARRIOR:
+				iconPath += "garrosh";
+				break;
+			default:
+			case ANY:
+				iconPath += "unknown";
+				break;
+	
+			}
+			return iconPath + ".png";
+		} else {
+			String assetsFolder = UserHomeMetastone.getPath().replace("\\\\", "/") + "/" + "assets";
```

Comment:
Should probably use File.separator for compatibility with filesystems that use a file separator other than "/".  Ran into this problem when I refactored the CardCatalogue to load cards from the user's home metastone dir.